htdocs/projet/class/project.class.php

```diff
@@ -1589,13 +1590,13 @@ public function getProjectsAuthorizedForUser($user, $mode = 0, $list = 0, $socid
 		if ($mode == 0) {
 			$sql .= " AND ( p.public = 1";
 			$sql .= " OR ( ec.fk_c_type_contact IN (".$this->db->sanitize(join(',', array_keys($listofprojectcontacttype))).")";
-			$sql .= " AND ec.fk_socpeople = ".((int) $user->id).")";
+			$sql .= " AND ( ec.fk_socpeople = ".((int) $user->id)." ) OR ( ec.fk_socpeople = ".((int) $user->contact_id)." ))";
```

Comment:
Permission on Dolibarr are defined on a users and only users. If there is no user, it can't have permission allowed.
Link on project to contact is only informative.